jax/_src/lax/control_flow/loops.py

```diff
@@ -352,12 +352,30 @@ def _create_jaxpr(init):
     consts = [*new_consts, *consts]
     num_carry -= len(new_consts)
 
+  # When an extensive output is forwarded from an extensive input, we can
+  # avoid copying it by pruning it from the jaxpr and forwarding manually. We
+  # don't need to update the indexing based on the optimization above since it
+  # doesn't change the total number of consts and carries combined. But, we do
+  # remove the number of consts from the index since `_jaxpr_forwarding`
+  # includes the Jaxpr's constvars when indexing.
+  in_fwd = [in_idx - len(consts) if in_idx is not None and
```

Comment:
Instead of doing any re-indexing, maybe we should just recompute the forwards again after the above constification munging?